sdk/core/azure-core-amqp/src/models/message_source.cpp

```diff
@@ -24,7 +27,115 @@ namespace Azure { namespace Core { namespace _internal {
 
 namespace Azure { namespace Core { namespace Amqp { namespace Models { namespace _internal {
 
+  MessageSource::~MessageSource(){};
+
   MessageSource::MessageSource(Models::AmqpValue const& source)
+      : m_impl{std::make_unique<_detail::MessageSourceImpl>(source)}
+  {
+  }
+
+  /* Note: This constructor should NOT be marked as explicit, because we want to enable the
+   * implicit construction of the MessageSource from a string - this allows callers to construct
+   * Link, MessageSender, and MessageReceiver objects without forcing the creation of a
+   * MessageSource object. */
+  MessageSource::MessageSource(std::string const& address)
+      : m_impl{std::make_unique<_detail::MessageSourceImpl>(address)}
+  {
+  }
+
+  /* Note: This constructor should NOT be marked as explicit, because we want to enable the
```

Comment:
It is the header file where things are declared explicit, so I think .hpp file is the netter place for this comment. I am not proposing to turn it into a doxygen comment.